### ISSUE TITLE
Fail to build with gcc-7

### ISSUE THREAD

**Issue by guikcd** [inspircd/inspircd#1309]
Hi,

inspircd 2.0.23 fail to build with gcc 7:

```
$ which g++
/usr/bin/g++
[...]
# Really launch configure
./configure --disable-interactive \
        --with-cc=g++ \
        --uid=39 \
        --enable-gnutls \
        --prefix=/usr/lib/inspircd \
        --config-dir=/etc/inspircd \
        --module-dir=/usr/lib/inspircd/modules \
        --binary-dir=/usr/sbin
`g++` was not found! A C++ compiler is required to build InspIRCd!
[...]
```

The `configure` script is looking for minor version of gcc (https://github.com/inspircd/inspircd/blob/insp20/configure#L353-L354) but gcc-7 has only a major version for now:

```
$ g++ -dumpversion
7
$ g++ --version
g++ (Debian 7-20170316-1) 7.0.1 20170316 (experimental) [trunk revision 246203]
[...]
```

There is a new option introduced with gcc-7:

```
$ g++ -dumpfullversion
7.0.1
```

I think the variable `$config{GCCMINOR}` seems not quite useful... maybe it is better to not rely on minor version of gcc.

Sources:
https://gcc.gnu.org/ml/gcc-patches/2017-01/msg00851.html
https://gcc.gnu.org/ml/gcc-patches/2017-01/msg00593.html

**Comment by SaberUK** [comment]
I don't really want to get rid of GCCMINOR because thats a very invasive change for 2.0 especially considering that this abomination has been rewritten entirely for 3.0.

Speculative fix: https://github.com/inspircd/inspircd/compare/insp20...SaberUK:insp20+dumpver

**Comment by SaberUK** [comment]
This should be fixed now #1318 has been merged.
